src/ClientData/include/ClientData/CallstackType.h

```diff
@@ -7,17 +7,37 @@
 
 #include <cstdint>
 
-#include "ClientProtos/capture_data.pb.h"
+#include "GrpcProtos/capture.pb.h"
 
 namespace orbit_client_data {
 
 using ThreadID = uint32_t;
 
-[[nodiscard]] std::string CallstackTypeToString(
-    orbit_client_protos::CallstackInfo::CallstackType callstack_type);
-
-[[nodiscard]] std::string CallstackTypeToDescription(
-    orbit_client_protos::CallstackInfo::CallstackType callstack_type);
+// This enum represents the different (error) types of a callstack. `kComplete` represents and
+// intact callstack, while the other enum values describe different errors. In addition to the
+// values in `orbit_grpc_protos::Callstack`, error types detected by the client are added.
+// The enum must be kept in sync with the one in `orbit_grpc_protos::Callstack`.
+enum class CallstackType {
+  kComplete = 0,
+  kDwarfUnwindingError = 1,
+  kFramePointerUnwindingError = 2,
+  kInUprobes = 3,
+  kInUserSpaceInstrumentation = 7,
+  kCallstackPatchingFailed = 4,
+  kStackTopForDwarfUnwindingTooSmall = 5,
+  kStackTopDwarfUnwindingError = 6,
+
+  // These are set by the client and are in addition to the ones in
+  // Callstack::CallstackType.
+  kFilteredByMajorityOutermostFrame = 100
```

Comment:
I don't think you need to specify the values like for the proto, since `GrpcCallstackTypeToCallstackType` uses a `switch` for the translation.